src/main/java/teammates/common/datatransfer/attributes/StudentAttributes.java

```diff
@@ -366,11 +366,12 @@ public StudentAttributes build() {
     }
 
     /**
-     * Helper class to specific the fields to update in {@link StudentAttributes}.
+     * Helper class to specify the fields to update in {@link StudentAttributes}.
      */
     public static class UpdateOptions {
         private String courseId;
         private String email;
+        private boolean shouldRegenerateRegistrationKey;
```

Comment:
`UpdateOptions` is used to update the attributes of entity.

Obviously, regkey is not something that can be updated (with a new value carry by the UpdateOptions).

How about put the regeneration logic in the `studentDb`?